src/book.py

```diff
@@ -560,22 +561,99 @@ def _read_kraken_ledgers(self, file_path: Path) -> None:
+                    held_operations.append({
+                        "refid": refid,
+                        "operation": operation,
+                        "utc_time": utc_time,
+                        "platform": platform,
+                        "change": change,
+                        "coin": coin,
+                        "row": row,
+                        "file_path": file_path,
+                        "fee": fee,
+                    })
+
+                    if operation == "Deposit":
+                        # append only the second deposit to the operations list
+                        if refid_idxs:
+                            append_operation = True
+                        else:
+                            append_operation = False
+                    elif operation == "Withdrawal":
+                        # Append the first withdrawal to the operations list as soon
+                        # as the second withdrawal occurs. Therefore, for the second
+                        # withdrawal, overwrite the variables with the data of the first
+                        # withdrawal and append this to the operations list.
+                        if refid_idxs:
+                            append_operation = True
+                            # required for type annotation: convert Optional[str] to str
+                            operation = str(held_operations[idx]["operation"])
+                            utc_time = held_operations[idx]["utc_time"]
+                            platform = held_operations[idx]["platform"]
+                            change = held_operations[idx]["change"]
+                            coin = held_operations[idx]["coin"]
+                            row = held_operations[idx]["row"]
+                            file_path = held_operations[idx]["file_path"]
+                            fee = held_operations[idx]["fee"]
+                        else:
+                            append_operation = False
+
+                if append_operation:
                     self.append_operation(
-                        "Fee", utc_time, platform, fee, coin, row, file_path
+                        operation, utc_time, platform, change, coin, row, file_path
                     )
-
-        assert dup_state["deposit"] == 0, (
-            "Orphaned deposit. (Must always come in pairs). " "Is your file corrupted?"
-        )
-        assert dup_state["withdrawal"] == 0, (
-            "Orphaned withdrawal. (Must always come in pairs). "
-            "Is your file corrupted?"
-        )
+                    if fee != 0:
+                        self.append_operation(
+                            "Fee", utc_time, platform, fee, coin, row, file_path
+                        )
+                    if operation == "StakingInterest":
+                        # the rewarded coins are added to the staked portfolio
+                        # TODO: directly stake the rewarded coins, not just any coin
```

Comment:
Why should the rewarded coins directly be staked? Does Kraken work like this? Please write this as comment. I find this none trivial.

I am unsure about this, but staking might follow FIFO. So the current behavior is correct and the TODO is unnecessary.

Reply:
Yes, the way it works with Kraken is that the rewarded coins are directly added to the staked balance. Assuming that spot and staking are considered two separate "wallets", this is different from adding the rewards to the spot account and then moving the same amount to staking, because FIFO (staking of an older spot coin) would mess up the tax calculation.
That's why I added the todo. Following our discussion in https://github.com/provinzio/CoinTaxman/issues/57, it would mean that we have to add the rewards directly to the (virtual) `kraken-staked` exchange.

Reply:
Ah ok, that does make sense.

Keep in mind, that this should only be done when `MULTI_DEPOT` is activate. We might want to write this to the TODO.

Reply:
good point, I'll add the comments